src/main/java/com/ldtteam/blockout/BOScreen.java

```diff
@@ -111,44 +114,49 @@ public BOScreen(final Window w)
         window = w;
     }
 
-    public static double getScale()
-    {
-        return scale;
-    }
-
-    private static void setScale(final Minecraft mc)
-    {
-        // Seems to work without the sides now
-        // Failsave
-        if (mc != null)
-        {
-            scale = mc.mainWindow.getGuiScaleFactor();
-        }
-    }
-
     @Override
+    @SuppressWarnings("deprecation")
     public void render(final MatrixStack ms, final int mx, final int my, final float f)
     {
         if (window.hasLightbox() && super.minecraft != null)
         {
             super.renderBackground(ms);
         }
 
-        setScale(minecraft);
-        final float oldZ = Minecraft.getInstance().getItemRenderer().zLevel;
-        Minecraft.getInstance().getItemRenderer().zLevel = MatrixUtils.getLastMatrixTranslateZ(ms);
+        final double fbHeight = minecraft.mainWindow.getFramebufferHeight();
+        final double fbWidth = minecraft.mainWindow.getFramebufferWidth();
+        final double guiWidth = Math.max(fbWidth, 320.0d);
+        final double guiHeight = Math.max(fbHeight, 240.0d);
 
-        ms.push();
-        ms.translate(x, y, 0);
-        window.draw(ms, mx - x, my - y);
-        ms.pop();
+        final float renderZlevel = MatrixUtils.getLastMatrixTranslateZ(ms);
+        final float oldZ = minecraft.getItemRenderer().zLevel;
+        minecraft.getItemRenderer().zLevel = renderZlevel;
+
+        mcScale = minecraft.mainWindow.getGuiScaleFactor();
+        renderScale = window.getRenderType().calcRenderScale(minecraft.mainWindow, window);
+
+        x = Math.floor((guiWidth - window.getWidth() * renderScale) / 2.0d);
+        y = Math.floor((guiHeight - window.getHeight() * renderScale) / 2.0d);
+
+        RenderSystem.matrixMode(GL11.GL_PROJECTION);
```

Comment:
Not sure this is needed here.

Reply:
Hmm?

Reply:
It at least requires some documentation.

Reply:
Literally basic opengl rendering setup, but ok

Reply:
No why is it here.
Mc already applies this when rendering uis
So why do you need to reset everything again?
That is what I am wondering

Reply:
It's not resetting, you just replace projection with unscaled one and do the scaling in modelview stack, you can probably set this up also in modelview stack but this is way safer

Reply:
Like you can do in modelview stack
ms.getlast.getmatrix.loadIdentity
ms.translate zLevel
ms.scale renderScale/mcScale
but you would need to fix scissors since they would still be in mcScale environment rather than in framebuffer scaled environment